server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -967,20 +973,19 @@ private static void updateRemoteStoreSettings(Settings.Builder settingsBuilder,
                 return;
             }
 
-            // Verify REPLICATION_TYPE cluster level setting is not conflicting with Remote Store
-            if (INDEX_REPLICATION_TYPE_SETTING.exists(requestSettings) == false
-                && CLUSTER_REPLICATION_TYPE_SETTING.get(clusterSettings).equals(ReplicationType.DOCUMENT)) {
+            // Verify replication type of index is not DOCUMENT
```

Comment:
Nit: Since the only alternate to docrep is segrep maybe specify that this check is to verify if index has the correct replication type (segment replication)